library/Fission/Web/Auth.hs

```diff
@@ -1,30 +1,30 @@
 module Fission.Web.Auth
-  ( server
-  , context
+  ( Checks
+  , server
   , basic
   , user
   , checkUser
   ) where
 
 import Crypto.BCrypt
-import Database.Selda
+import Database.Esqueleto
 import Servant
 
+import Fission.Models
 import Fission.Prelude
-import Fission.Storage.Query
-import Fission.User as User hiding (username)
-import Fission.Web.Server
+
+type Checks = '[BasicAuthCheck (Entity User), BasicAuthCheck ByteString]
 
 server
-  :: HasServer api '[BasicAuthCheck User, BasicAuthCheck ByteString]
+  :: HasServer api Checks
   => Proxy api
-  -> cfg
-  -> RIOServer cfg api
-  -> Server api
-server api cfg = hoistServerWithContext api context (toHandler cfg)
-
-context :: Proxy (BasicAuthCheck User ': BasicAuthCheck ByteString ': '[])
-context = Proxy
+  -> (forall a . m a -> Handler a)
+  -> ServerT api m
+  -> ServerT api Handler
+server api = hoistServerWithContext api context
+  where
+    context :: Proxy Checks
+    context = Proxy
```

Comment:
I reworked this a bit, to make it work more nicely with the `Monad*` capability style.